pilot/pkg/serviceregistry/kube/controller/ambient/multicluster.go

```diff
@@ -0,0 +1,549 @@
+		filter,
+		"Gateways",
+		opts,
+		nil,
+	), opts.WithName("AllGatewaysInformers")...)
+	gatewayInformersByCluster := informerIndexByCluster(allGateways)
+	globalGateways := krt.NewCollection(clusters, collectionFromCluster[*v1beta1.Gateway]("Gateways", gatewayInformersByCluster, opts))
+	globalGatewaysByCluster := nestedCollectionIndexByCluster(globalGateways)
+
+	allGatewayClasses := krt.NewCollection(clusters, informerForCluster[*v1beta1.GatewayClass](
+		filter,
+		"GatewayClasses",
+		opts,
+		nil,
+	), opts.WithName("AllGatewayClassesInformers")...)
+	gatewayClassInformersByCluster := informerIndexByCluster(allGatewayClasses)
+	globalGatewayClasses := krt.NewCollection(clusters, collectionFromCluster[*v1beta1.GatewayClass]("GatewayClasses", gatewayClassInformersByCluster, opts))
+	globalGatewayClassesByCluster := nestedCollectionIndexByCluster(globalGatewayClasses)
+
+	allNamespaces := krt.NewCollection(clusters, informerForCluster[*v1.Namespace](
+		filter,
+		"Namespaces",
+		opts,
+		nil,
+	), opts.WithName("AllNamespacesInformers")...)
+	namespaceInformersByCluster := informerIndexByCluster(allNamespaces)
+	globalNamespaces := krt.NewCollection(clusters, collectionFromCluster[*v1.Namespace]("Namespaces", namespaceInformersByCluster, opts))
+	globalNamespacesByCluster := nestedCollectionIndexByCluster(globalNamespaces)
+
+	allEndpointSlices := krt.NewCollection(clusters, informerForCluster[*discovery.EndpointSlice](
+		filter,
+		"EndpointSlices",
+		opts,
+		nil,
+	), opts.WithName("AllEndpointSlices")...)
+	endointSliceInformersByCluster := informerIndexByCluster(allEndpointSlices)
+	globalEndpointSlices := krt.NewCollection(clusters, collectionFromCluster[*discovery.EndpointSlice]("EndpointSlice", endointSliceInformersByCluster, opts))
+	globalEndpointSlicesByCluster := nestedCollectionIndexByCluster(globalEndpointSlices)
+
+	allNodes := krt.NewCollection(clusters, informerForCluster[*v1.Node](
+		filter,
+		"Nodes",
+		opts,
+		func(c Cluster) kclient.Filter {
+			return kclient.Filter{
+				ObjectFilter:    c.Client.ObjectFilter(),
+				ObjectTransform: kubeclient.StripNodeUnusedFields,
+			}
+		},
+	), opts.WithName("AllNodes")...)
+	nodeInformersByCluster := informerIndexByCluster(allNodes)
+	globalNodes := krt.NewCollection(clusters, collectionFromCluster[*v1.Node]("Nodes", nodeInformersByCluster, opts))
+	// Set up collections for remote clusters
+	GlobalNetworks := buildGlobalNetworkCollections(
+		clusters,
+		globalGatewaysByCluster,
+		options,
+		opts,
+	)
+	a.networks = GlobalNetworks
+	GlobalWaypoints := GlobalWaypointsCollection(
+		clusters,
+		globalGatewaysByCluster,
+		globalGatewayClassesByCluster,
+		globalPodsByCluster,
+		GlobalNetworks.SystemNamespaceNetworkByCluster,
+		opts,
+	)
+	// We need to get the waypoints for the local cluster, so we need to wait for the
+	// collection to be synced
+	if !GlobalWaypoints.WaitUntilSynced(opts.Stop()) {
```

Comment:
I don't see how this can work? We don't start the kube client until after we call this...?

Reply:
Yeah I ran into this locally while writing tests; I'll push when it's fixed